conda_smithy/configure_feedstock.py

```diff
@@ -881,14 +994,22 @@ def migrate_combined_spec(combined_spec, forge_dir, config, forge_config):
 
 
 def _conda_build_api_render_for_smithy(
-    recipe_path,
-    config=None,
-    variants=None,
-    permit_unsatisfiable_variants=True,
-    finalize=True,
-    bypass_env_check=False,
+    recipe_path: str,
+    config = None,
+    variants: Optional[
+        Union[
+            Dict[str, Union[List[str], Dict[str, Dict[str, str]]]],
+            Dict[
+                str,
+                Union[List[str], Dict[str, Dict[str, str]], List[List[str]]],
+            ],
+        ]
+    ] = None,
```

Comment:
I think this whole type simplifies to `Optional[Dict[str, Union[List[str], List[List[str]], Dict[str, Dict[str, str]]]]]`.